R/eml_to_spice.R

```diff
@@ -0,0 +1,294 @@
+
+
+  out <- biblio_eml[, fields[fields %in% colnames(biblio_eml)]]
+
+  if(!is.null(path)){
+    if(!dir.exists(path)){
+      dir.create(path)
+    }
+    readr::write_csv(out, file.path(path, "biblio.csv"))
+  }
+
+  return(out)
+}
+
+#' Get creators from EML
+#'
+#' Return EML creators in the dataspice creators.csv format.
+#'
+#' @param eml (emld) an EML object
+#' @param path (character) folder path for saving the table to disk
+#'
+#' @importFrom purrr discard
+#' @importFrom tibble enframe
+#' @importFrom tidyr spread
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' eml_path <- system.file("example-dataset/broodTable_metadata.xml", package = "dataspice")
+#' eml <- read_eml(eml_path)
+#' es_creators(eml)
+#' }
+
+es_creators <- function(eml, path = NULL) {
+  people <- get_entities(eml,
+                         entities = c("creator", "contact", "associatedParty", "metadataProvider"),
+                         level_id = c("individualName", "organizationName"))
+  if(!is.null(names(people))){
+    people <- people[names(people) == ""]
+  }
+
+  people_parsed <- lapply(people, function(x){x %>%
+      unlist() %>%
+      tibble::enframe() %>%
+      dplyr::mutate(name = dplyr::case_when(
+        grepl("userId.userId", name) ~ "id",
+        grepl("givenName", name) ~ "givenName",
+        grepl("surName", name) ~ "familyName",
+        grepl("organizationName", name) ~ "affiliation",
+        grepl("electronicMailAddress", name) ~ "email"
+      )) %>%
+      stats::na.omit() %>%
+      # merge fields together if duplicated (ex: givenName1 & givenName2)
+      group_by(name) %>%
+      dplyr::summarize(value = paste(value, collapse = " ")) %>%
+      tidyr::spread(name, value)
+  })
+
+  out <- dplyr::bind_rows(people_parsed) %>%
+    dplyr::distinct()
+
+  fields <- c("id", "givenName", "familyName", "affiliation", "email")
+  out <- out[, fields[fields %in% colnames(out)]]
+
+  if(!is.null(path)){
+    if(!dir.exists(path)){
+      dir.create(path)
+    }
+    readr::write_csv(out, file.path(path, "creators.csv"))
+  }
+
+  return(out)
+}
+
+#' Get dataspice tabular formats from EML
+#'
+#' Return EML in the dataspice dataframes.
+#'
+#' @param eml (emld) an EML object
+#' @param path (character) folder path for saving the table to disk
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' eml_path <- system.file("example-dataset/broodTable_metadata.xml", package = "dataspice")
+#' eml <- read_eml(eml_path)
+#' my_spice <- eml_to_spice(eml, ".")
+#' }
+
+eml_to_spice <- function(eml, path = NULL) {
+  out <- list(attributes = es_attributes(eml, path),
+              access = es_access(eml, path),
+              biblio = es_biblio(eml, path),
+              creators = es_creators(eml, path))
+
+  invisible(out)
```

Comment:
Any reason why this is `invisible`?

Reply:
Mostly because it's likely really long and probably not worth printing to the console automatically.  Though if you think people would expect it, then I can changed to `return` so that it's more intuitive.  Alternatively, I can also return a condensed version/a message/etc

Reply:
Fair enough. I'd say that's fine, then. I don't think I've run into this enough to have a strong feeling on it.